src/Logger-MCC_DAQ/main.py

```diff
@@ -27,37 +27,43 @@
 # Contribution: We like to thank TU Dresden/Shayan Miri for providing the initial version of this driver.
 
 # SweepMe! device class
-# Type: Logger
-# Device: Measurement Computing Corporation DAQ devices
+# *Module: Logger
+# *Device: Measurement Computing Corporation DAQ devices
 
 from pysweepme import addFolderToPATH
 from pysweepme.EmptyDeviceClass import EmptyDevice
 
 addFolderToPATH()
 
-# this driver needs libraries installed by the manufacturer software InstaCal
+# this driver needs libraries installed by the manufacturer software package Universal Library™
+# The ImportError is raised in the find_ports method to allow loading the driver in the GUI
+mcculw_library_missing = False
 try:
     from mcculw import ul
     from mcculw.device_info import DaqDeviceInfo
     from mcculw.enums import AnalogInputMode, InterfaceType, ULRange
-except FileNotFoundError as e:
-    msg = "MCC DAQ Software missing. Install InstaCal and Universal Library (UL)."
-    raise ImportError(msg) from e
+except FileNotFoundError:
+    mcculw_library_missing = True
 
 
 class Device(EmptyDevice):
+    """Driver to read out MCC DAQ devices."""
     description = """
                 <h4>MCC High-Speed Multifunction DAQ</h4>
 
-                <p>To use this driver, installation of MCC DAQ Software, including Universal Library™ is needed.
+                <p>To use this driver, installation of Universal Library™ from the MCC DAQ Software package is needed.
                 Please download it from the <a href="https://www.mccdaq.com/Software-Downloads">MCC Homepage</a></p>
 
                 <p>If your device supports additional AI ranges, they can be added by extending the
                 <code>available_ai_ranges</code> dictionary.</p>
                 """
 
-    def __init__(self):
+    def __init__(self) -> None:
+        """Initialize driver parameters."""
         EmptyDevice.__init__(self)
+        if mcculw_library_missing:
```

Comment:
throwing an error in __init__ works but sometimes leads to the problem that the error message pops up to often, e.g. when the the driver is the first item in the device list of a module and one clicks on the combo box of the field "Device".

An alternative would be to add the check at the beginning of "find_ports" and at the beginning of "connect" so just before the library is really needed. Then, the user is immediately informed when searching for devices/ports or when a measurement is started